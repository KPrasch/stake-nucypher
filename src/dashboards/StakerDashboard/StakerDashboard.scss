
@media (min-width: 576px) and (max-width: 992px) {
  .tokens-row {
    [class^='col'] {
      margin-top: 20px;
    }
  }
}

.number-of-tokens {
  display: inline-block;
}
