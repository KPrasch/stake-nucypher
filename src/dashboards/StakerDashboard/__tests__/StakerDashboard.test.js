import React from 'react';
import StakerDashboard from '../StakerDashboard';
import { render, fireEvent, wait } from '@testing-library/react';
jest.mock('../../../web3Initializer');

describe('<StakerDashboard />', () => {

  const originalDate = Date.now;
  beforeAll(() => {
    Date.now = () => 1582581721708;
  });

  afterAll(() => {
    Date.now = originalDate;
  });

  it('should render correctly', async () => {
    await wait(() => {
      const { asFragment } = render(<StakerDashboard />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

});
