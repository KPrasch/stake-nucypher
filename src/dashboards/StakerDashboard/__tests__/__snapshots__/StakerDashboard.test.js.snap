// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StakerDashboard /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="panel row"
  >
    <div
      class="col"
    >
      <h2
        class="text-center"
      >
        Welcome back, 
      </h2>
      <div
        class="mt-5 row"
      >
        <div
          class="col"
        >
          <p
            class="h6 text-center"
          >
            Current period
          </p>
          <p
            class="h4 text-center"
          >
            18945
          </p>
        </div>
        <div
          class="col"
        >
          <p
            class="h6 text-center"
          >
            Next period in
          </p>
          <p
            class="h4 text-center"
          >
            1 hours
          </p>
        </div>
      </div>
      <div
        class="mt-5 tokens-row row"
      >
        <div
          class="col"
        >
          <p
            class="h6 text-center"
          >
            Tokens staked
          </p>
          <p
            class="h4 text-center"
          >
             
            <br />
            NU
          </p>
          <div
            class="action d-flex justify-content-center"
          >
            <button
              class="button-action mt-2 btn btn-primary"
              type="button"
            >
              Add stake
            </button>
          </div>
        </div>
        <div
          class="col"
        >
          <p
            class="h6 text-center"
          >
            Total rewards
          </p>
          <p
            class="h4 text-center"
          >
             
            <br />
            NU
          </p>
        </div>
        <div
          class="col"
        >
          <p
            class="h6 text-center"
          >
            Available for withdrawal
          </p>
          <p
            class="h4 text-center"
          >
             
            <br />
            NU
          </p>
          <div
            class="action d-flex justify-content-center"
          >
            <button
              class="button-action mt-2 btn btn-secondary"
              type="button"
            >
              Withdraw
            </button>
          </div>
        </div>
      </div>
      <div
        class="mt-5 row"
      />
      <div
        class="mt-5 row"
      />
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col"
    />
  </div>
</DocumentFragment>
`;
